### PR TITLE
Mesh 542/current did refactor

### PR DESCRIPTION
**Pull request by fmiguelgarcia**
New `Context` helper class.
It simplifies the access to current identity (even in test) to the following pattern:
```
let from_did = Context::current_identity_or::<Self>(&sender_key)?;
```

### CODE REVIEW

**Comment by maxsam4** [comment]
Was this based on a very old branch? I see a lot of commits that don't relate to this PR. Can you please rebase on top of the latest develop?

File: pallets/identity/src/lib.rs [hunk 1]
```diff
@@ -542,16 +542,8 @@ decl_module! {
...
-                None => {
-                    if let Some(did) = Self::get_identity(&sender_key) {
-                        did
-                    } else {
-                        return Err(Error::<T>::NoDIDFound.into());
-                    }
-                }
-            };
+            let from_did = Context::current_identity_or::<Self>(&sender_key)
+                .ok_or_else(|| DispatchError::from(Error::<T>::NoDIDFound))?;
```
Comment (maxsam4) [inline pallets/identity/src/lib.rs]: Will it make sense to add a function in `Context` that returns did if it exists or else it returns `DispatchError::from(Error::<T>::NoDIDFound)`? that way, we can reduce all these calls to 
`let from_did = Context::current_identity_or_error::<Self>(&sender_key)?;`

**Comment by fmiguelgarcia** [comment]
@maxsam4 
About commits, yes It was base on a old branch, but I didn't expect that PR shows all of them.
About using `current_identity`, you are right. There is another task for that (I am working on that one right now).

File: pallets/identity/src/lib.rs (hunk 1 above)
Comment (fmiguelgarcia) [reply pallets/identity/src/lib.rs]: Done.

**Pull request merged**